srv/modules/runners/rescinded.py

```diff
@@ -68,6 +74,45 @@ def osds(cluster='ceph'):
             log.debug(data[minion])
     return list(set(_ids))
 
+
+def orphaned_hosts_list(cluster='ceph'):
+    """
+    List any orphaned host entries in the CRUSH map
+    """
+    caller = salt.client.Caller()
+    data = caller.function('osd.orphaned')
+    assert isinstance(data, list), "List of orphaned CRUSH host entries is not a list"
+    return data
+
+
+def orphaned_hosts(cluster='ceph'):
+    """
+    Return true if there are any orphaned host entries in the CRUSH map
+    """
+    if len(orphaned_hosts_list()) > 0:
+        return True
+    else:
+        return False
+
+
+def delete_orphaned_crushhosts(cluster='ceph'):
+    """
+    If hosts tagged for deletion by ids() are orphaned in the CRUSH map, remove
+    them (best effort)
+    """
+    caller = salt.client.Caller()
+    __opts__ = salt.config.client_config('/etc/salt/master')
+    __grains__ = salt.loader.grains(__opts__)
+    orphaned_hosts = orphaned_hosts_list()
+    crushhosts_to_delete = __grains__['crushhosts_to_delete']
```

Comment:
could I use `caller.function('grain.get', ['crushhosts_to_delete'])` to retrieve the grain value, and dispense with `__opts__` and `__grains__`?